libs/chain-signatures/contract/src/lib.rs

```diff
@@ -809,14 +792,7 @@ impl VersionedMpcContract {
         #[callback_unwrap] signature: SignatureResult<SignatureResponse, SignaturePromiseError>,
     ) -> Result<SignatureResponse, Error> {
         match self {
```

Comment:
What is the purpose of the outer `match self` here?

Reply:
I believe the original author of the contract intended to emphasize that this function is part of the V0 logic and wanted to ensure its behavior does not accidentally get changed.

Because this function is only called via `env::promise_then(..., "return_signature_on_finish", ...)`, searching for references within an IDE might not yield any results and the V0s dependency  on it might get easily overlooked.

Personally, I found this explicit style helpful for familiarizing myself with the contract code, which is why I chose to keep it here.